### PR TITLE
Mono update

### PR DESCRIPTION
**Pull request by chris-sanders**
This is built on top of MR #4 and switches to the mono-project repository for Mono. A functional test was added that checks that the Mono version is newer than 5.x. This also includes the new upgrade testing, which verifies that packages are upgraded if the current charm store version is installed and then upgraded.

This should close Issue #3 although I've manually applied the fix to my deploy so I won't be able to verify this fixes with just a charm upgrade as expected.

### CODE REVIEW

**Comment by chris-sanders** [comment]
Actually, I seem to have missed something in the upgrade, patch incoming.

**Comment by chris-sanders** [comment]
Now it's working, somehow I tested with already deployed containers previously but this was a clean run and the upgrade hook is now used to trigger the update.

File: lib/lib_radarr.py [hunk 1]
```diff
@@ -27,10 +28,11 @@ def __init__(self):
         self.config_file = self.config_dir + '/config.xml'
         self.service_name = 'radarr.service'
         self.service_file = '/etc/systemd/system/' + self.service_name
+        self.kv = unitdata.kv()
         self.deps = [
             'libmono-cil-dev',
             'curl',
-            'mediainfo'
+            'mediainfo',
```
Comment (digitalrane) [inline lib/lib_radarr.py:35]: I don't like or understand this but I'll allow it

File: lib/lib_radarr.py [hunk 2]
```diff
@@ -150,6 +152,10 @@ def setup_plex(self, hostname, port, user=None, passwd=None):
         host.service_start(self.service_name)
 
     def install_deps(self):
+        fetch.add_source("deb https://download.mono-project.com/repo/ubuntu stable-{series} main",
+                         key="3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF")
```
Comment (digitalrane) [inline lib/lib_radarr.py:156]: We should move this to config.
If we really wanted to shave some yaks, we could create layer-mono... then move layer-sonarr to that, too.

File: tests/functional/conftest.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    await _model.disconnect()
+    if not os.getenv('PYTEST_KEEP_MODEL'):
+        await controller.destroy_model(model_name)
+        while model_name in await controller.list_models():
+            await asyncio.sleep(1)
+
+
+@pytest.fixture(scope='module')
+async def jujutools(controller, model):
+    tools = JujuTools(controller, model)
```
Comment (digitalrane) [inline tests/functional/conftest.py:65]: This is so tidy!

**Review by digitalrane — APPROVED** [review]
A couple of comments which are really just things we should discuss about future organisation of Mono-based charms, but LGTM

**Pull request merged**